### DISCUSSION TITLE
`gopass sync` works without ssh key in ssh agent

### DISCUSSION THREAD

**Discussion by revengel** [gopasspw/gopass#2258] [General]
### Summary

`gopass sync` command works without ssh key in ssh agent

Please help me to understand the issue

### Steps To Reproduce

* clear `ssh-agent` - `ssh-add -D`
* adding ssh key in `ssh-agent` for gopass syncing `ssh-add /path/to/private/ssh-key`
* try to generate some password `gopass generate some/password/path1/password`
* make `gopass sync`
* password has been generated and pushed to git-server
* clear `ssh-agent` - `ssh-add -D`
* There are no any ssh-key in ssh-agent now. Synchronization should not work now.
* try to generate another password `gopass generate some/password/path2/password`
* make `gopass sync`
* But now the sync has happened again and the new password has been pushed to git server
* At the same time `git clone`, `git pull`, `git push` fall with an error `Please make sure you have the correct access rights`
* ???

### Version

`gopass 1.14.3-git+HEAD go1.18.2 linux amd64`

**Comment by dominikschulz** [comment]
Yes, that does make sense. We use an persistent ssh control Socket to speed up subsequent sync operations: https://github.com/gopasspw/gopass/blob/master/internal/backend/storage/gitfs/ssh_others.go

You can disable that by overriding GIT_SSH_COMMAND.

**Comment by cdarken** [comment]
I need a bit of help with this. For me gopass just hangs when it's running this command in the background. I have to kill the process. How can I debug what is going on? GOPASS_DEBUG env var is set.

**Reply by dominikschulz** [reply]
Did you set `GOPASS_DEBUG_LOG=/tmp/gopass.log` (or simliar)?
That log is pretty verbose but should give you a good idea what's happening.

**Reply by cdarken** [reply]
Thanks for the answer. I tried just now setting that var too and tailing it to watch the output.
It appears to just hang while running `git pull origin master`. But if I kill it and run again it works.
If I set GIT_SSH_COMMAND to just "ssh" would that fix it for good? Maybe the issue is caused by the persistent connection (the socket is created in /tmp, btw).

**Reply by dominikschulz** [reply]
`GIT_SSH_COMMAND=""` should work.

**Reply by cdarken** [reply]
I tried but my sync wouldn't work without it. It seems that I have a problem with ssh persistent connection on first run only, can't find a solution for it. I'll just set the env var to "ssh", it seems to be working fine. Thanks for helping and for this nice software!
